File: Backend/src/FindYourClinic.Domain/Entities/AiChatMessage.cs
namespace FindYourClinic.Domain.Entities;

public class AiChatMessage
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string UserId { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: Mobile/lib/features/ai_health/domain/entities/ai_chat_message.dart
// Domain entity for AI chat messages.
// Pure Dart — no Flutter imports.

class AiChatMessage {
  final String role; // "user" or "assistant"
  final String content;
  final DateTime createdAt;

  const AiChatMessage({
    required this.role,
    required this.content,
    required this.createdAt,
  });
}
